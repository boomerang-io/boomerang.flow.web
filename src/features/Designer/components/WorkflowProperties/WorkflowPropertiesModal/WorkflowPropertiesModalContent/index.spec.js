import React from "react";
import Inputs from ".";
import { fireEvent, waitFor } from "@testing-library/react";
const mockfn = jest.fn();

const props = {
  isEdit: true,
  input: {
    defaultValue: "dogs",
    description: "Tim property",
    key: "tim.property",
    label: "Tim Property",
    required: false,
    type: "text"
  },
  loading: false,
  updateInputs: mockfn,
  setShouldConfirmModalClose: mockfn,
  closeModal: mockfn,
  inputsName: [],
  workflowActions: { updateWorkflowInput: mockfn, createWorkflowInput: mockfn },
  updateWorkflowProperties: mockfn
};

describe("Inputs --- Snapshot Test", () => {
  it("Capturing Snapshot of Inputs", async () => {
    const { baseElement } = rtlReduxRender(<Inputs {...props} />);
    expect(baseElement).toMatchSnapshot();
    await waitFor(() => {});
  });
});

describe("Inputs --- RTL", () => {
  it("Change default value by type correctly", async () => {
    const { getByText, queryByTestId, getByTestId } = rtlReduxRender(<Inputs {...props} />);
    expect(queryByTestId("text-input")).toBeInTheDocument();

    const typeSelect = getByTestId("input-type");

    fireEvent.change(typeSelect, {target:{value:"b"}});
    fireEvent.click(getByText(/boolean/i));

    expect(queryByTestId("text-input")).not.toBeInTheDocument();
    expect(queryByTestId("toggle")).toBeInTheDocument();

    fireEvent.change(typeSelect, {target:{value:"are"}});
    fireEvent.click(getByText(/text area/i));

    expect(queryByTestId("toggle")).not.toBeInTheDocument();
    expect(queryByTestId("text-area")).toBeInTheDocument();

    fireEvent.change(typeSelect, {target:{value:"sel"}});
    fireEvent.click(getByText(/select/i));

    expect(queryByTestId("text-area")).not.toBeInTheDocument();
    expect(queryByTestId("select")).toBeInTheDocument();

    await waitFor(() => {});
  });

  it("Shouldn't save property without key, label and type defined", () => {
    const { findByText, getByPlaceholderText, getByLabelText, getByTestId } = rtlReduxRender(
      <Inputs {...props} isEdit={false} input={undefined} />
    );
    waitFor(() => expect(findByText(/create/i)).toBeDisabled());

    const keyInput = getByPlaceholderText("key.value");
    const labelInput = getByPlaceholderText(/name/i);
    const typeSelect = getByLabelText(/type/i);

    fireEvent.change(keyInput, { target: { value: "test" } });
    fireEvent.change(labelInput, { target: { value: "test" } });

    fireEvent.change(typeSelect, {target:{value:"b"}});
    fireEvent.click(getByText(/boolean/i));

    waitFor(() => expect(findByText(/create/i)).toBeEnabled());

    await waitFor(() => {});
  });
});
