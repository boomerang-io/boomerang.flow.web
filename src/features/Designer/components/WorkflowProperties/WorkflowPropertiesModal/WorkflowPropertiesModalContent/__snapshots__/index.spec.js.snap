// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs --- Snapshot Test Capturing Snapshot of Inputs 1`] = `
<body
  id="app"
>
  <div>
    <form
      class="bx--bmrg-modal-flow-form"
    >
      <div
        class="bx--modal-content container"
      >
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="label"
            >
              Name
            </label>
            <div
              class="bx--text-input__field-wrapper"
            >
              <input
                class="bx--text-input bx--text__input"
                id="label"
                placeholder="Name"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="description"
            >
              Description
            </label>
            <div
              class="bx--text-input__field-wrapper"
            >
              <input
                class="bx--text-input bx--text__input"
                id="description"
                placeholder="Description"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-toggle --vertical"
        >
          <div>
            <div
              class="bx--bmrg-toggle__title"
            >
              <div
                class="bx--label"
                id="required-label"
                style="margin-bottom: 0px;"
              >
                Required
              </div>
            </div>
          </div>
          <div
            class="bx--form-item"
          >
            <input
              class="bx--toggle-input"
              id="required"
              type="checkbox"
            />
            <label
              aria-label="Toggle"
              class="bx--toggle-input__label"
              for="required"
            >
              <span
                class="bx--toggle__switch"
              >
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--off"
                />
                <span
                  aria-hidden="true"
                  class="bx--toggle__text--on"
                />
              </span>
            </label>
          </div>
        </div>
        <div
          class="bx--bmrg-select"
        >
          <div
            class="bx--list-box__wrapper"
          >
            <label
              class="bx--label"
              for="type"
            >
              Type
            </label>
            <div
              class="bx--combo-box bx--list-box"
              role="listbox"
              tabindex="-1"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Open menu"
                class="bx--list-box__field"
                data-toggle="true"
                role="button"
                tabindex="-1"
                type="button"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="type__menu"
                  aria-expanded="false"
                  aria-label="Choose an item"
                  autocomplete="off"
                  class="bx--text-input"
                  id="type"
                  placeholder="Select an item"
                  role="combobox"
                  value="Text"
                />
                <div
                  aria-label="Clear Selection"
                  class="bx--list-box__selection"
                  role="button"
                  tabindex="0"
                  title="Clear selected item"
                >
                  <svg
                    aria-hidden="true"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    style="will-change: transform;"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 4.7l-.7-.7L8 7.3 4.7 4l-.7.7L7.3 8 4 11.3l.7.7L8 8.7l3.3 3.3.7-.7L8.7 8z"
                    />
                  </svg>
                </div>
                <div
                  class="bx--list-box__menu-icon"
                >
                  <svg
                    aria-label="Open menu"
                    focusable="false"
                    height="16"
                    name="chevron--down"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    style="will-change: transform;"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
                    />
                    <title>
                      Open menu
                    </title>
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="defaultValue"
            >
              Default Value
            </label>
            <div
              class="bx--text-input__field-wrapper"
            >
              <input
                class="bx--text-input bx--text__input"
                data-testid="text-input"
                id="defaultValue"
                placeholder="Default Value"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="bx--modal-footer"
      >
        <button
          class="bx--btn bx--btn--secondary"
          tabindex="0"
          type="button"
        >
          Cancel
        </button>
        <button
          class="bx--btn bx--btn--primary"
          data-testid="inputs-modal-confirm-button"
          tabindex="0"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</body>
`;
